ddtrace/llmobs/utils.py

```diff
@@ -76,16 +80,85 @@ def __init__(self, messages: Union[List[Dict[str, str]], Dict[str, str], str]):
                 continue
             elif not isinstance(message, dict):
                 raise TypeError("messages must be a string, dictionary, or list of dictionaries.")
+
             content = message.get("content", "")
             role = message.get("role")
             if not isinstance(content, str):
                 raise TypeError("Message content must be a string.")
-            if not role:
-                self.messages.append(Message(content=content))
-                continue
-            if not isinstance(role, str):
-                raise TypeError("Message role must be a string, and one of .")
-            self.messages.append(Message(content=content, role=role))
+
+            msg_dict = Message(content=content)
+            if role:
+                if not isinstance(role, str):
+                    raise TypeError("Message role must be a string.")
+                msg_dict["role"] = role
+
+            tool_calls = message.get("tool_calls")
+            if tool_calls is not None:
+                if not isinstance(tool_calls, list):
+                    raise TypeError("tool_calls must be a list.")
+                formatted_tool_calls = []
+                for tool_call in tool_calls:
+                    if not isinstance(tool_call, dict):
+                        raise TypeError("Each tool_call must be a dictionary.")
+
+                    # name and arguments are required
+                    name = tool_call.get("name")
+                    arguments = tool_call.get("arguments")
+
+                    if not name or not isinstance(name, str):
+                        raise TypeError("ToolCall name must be a non-empty string.")
+                    if arguments is None or not isinstance(arguments, dict):
+                        raise TypeError("ToolCall arguments must be a dictionary.")
+
+                    formatted_tool_call = ToolCall(name=name, arguments=arguments)
+
+                    # Add optional fields if present
+                    tool_id = tool_call.get("tool_id")
+                    if tool_id and isinstance(tool_id, str):
+                        formatted_tool_call["tool_id"] = tool_id
+
+                    tool_type = tool_call.get("type")
+                    if tool_type and isinstance(tool_type, str):
+                        formatted_tool_call["type"] = tool_type
+
+                    formatted_tool_calls.append(formatted_tool_call)
```

Comment:
Can we extract this into a helper method? I think this Messages init is getting a bit long